pkg/cluster/sync.go

```diff
@@ -927,8 +909,8 @@ func (c *Cluster) syncRoles() (err error) {
 	}
 
 	// copy map for ProduceSyncRequests to include also system users
-	for userName, pgUser := range c.pgUsers {
-		newUsers[userName] = pgUser
```

Comment:
key must be the same as name of the user like it will be set for `dbUsers` list. Otherwise, ProduceSyncRequests will try to create the rotation user again, although it already exists.